ql/src/java/org/apache/hadoop/hive/ql/parse/type/HiveFunctionHelper.java

```diff
@@ -322,18 +324,24 @@ public RexNode getExpression(String functionText, FunctionInfo fi,
         // unix_timestamp(args) -> to_unix_timestamp(args)
         calciteOp = HiveToUnixTimestampSqlOperator.INSTANCE;
       }
-      expr = rexBuilder.makeCall(returnType, calciteOp, inputs);
+      expr = getExpression(returnType, calciteOp, inputs);
     }
 
     if (expr instanceof RexCall && !expr.isA(SqlKind.CAST)) {
       RexCall call = (RexCall) expr;
       expr = rexBuilder.makeCall(returnType, call.getOperator(),
           RexUtil.flatten(call.getOperands(), call.getOperator()));
     }
-
     return expr;
   }
 
+  private RexNode getExpression(RelDataType returnType, SqlOperator calciteOp, List<RexNode> inputs) {
+    if (calciteOp.getKind() == SqlKind.IN) {
+      return rexBuilder.makeIn(inputs.get(0), inputs.subList(1, inputs.size()));
+    }
+    return rexBuilder.makeCall(returnType, calciteOp, inputs);
+  }
+
```

Comment:
The code would be more uniform if instead of overloading the `getExpression` method we added the new `if clause` in the existing one.

Reply:
Yes, makes sense. I will update this.